components/constellation/constellation.rs

```diff
@@ -849,28 +849,38 @@ where
                     ipc_scheduler_receiver,
                 );
 
-                let (background_hang_monitor_sender, ipc_bhm_receiver) =
-                    ipc::channel().expect("ipc channel failure");
-                let background_hang_monitor_receiver =
-                    route_ipc_receiver_to_new_mpsc_receiver_preserving_errors(ipc_bhm_receiver);
+                let (background_hang_monitor_sender, background_hang_monitor_receiver) =
+                    if opts::get().background_hang_monitor {
+                        let (bhm_sender, ipc_bhm_receiver) =
+                            ipc::channel().expect("ipc channel failure");
+                        (
+                            Some(bhm_sender),
+                            Some(route_ipc_receiver_to_new_mpsc_receiver_preserving_errors(
+                                ipc_bhm_receiver,
+                            )),
+                        )
+                    } else {
+                        (None, None)
+                    };
 
                 // If we are in multiprocess mode,
                 // a dedicated per-process hang monitor will be initialized later inside the content process.
                 // See run_content_process in servo/lib.rs
-                let (background_monitor_register, sampler_chan) = if opts::multiprocess() {
-                    (None, vec![])
-                } else {
-                    let (sampling_profiler_control, sampling_profiler_port) =
-                        ipc::channel().expect("ipc channel failure");
-
-                    (
-                        Some(HangMonitorRegister::init(
-                            background_hang_monitor_sender.clone(),
-                            sampling_profiler_port,
-                        )),
-                        vec![sampling_profiler_control],
-                    )
-                };
+                let (background_monitor_register, sampler_chan) =
+                    if opts::multiprocess() | !opts::get().background_hang_monitor {
```

Comment:
Should this be `||` instead 👀?

```suggestion
                    if opts::multiprocess() || !opts::get().background_hang_monitor {
```

Reply:
Yeah, it should. Thanks for pointing it out. I'll do that now.